packages/aragon-wrapper/src/utils/apps.js

```diff
@@ -12,13 +12,28 @@ export const apmAppId = appName => namehash(`${appName}.aragonpm.eth`)
  * @return {Object} Method with radspec notice and function signature
  */
 export function findAppMethodFromData (app, data) {
-  if (app && app.functions) {
-    // Find the method
-    const methodId = data.substring(2, 10)
-    return app.functions.find(
-      (method) => keccak256(method.sig).substring(0, 8) === methodId
+  const methodId = data.substring(2, 10)
+  const { deprecatedFunctions, functions } = app || {}
+
+  let method
+
+  // First try to find the method in the current functions
+  if (Array.isArray(functions)) {
+    method = functions.find(
+      method => keccak256(method.sig).substring(0, 8) === methodId
+    )
+  }
+
+  // Couldn't find it in current functions, try on deprecated versions
+  if (!method && Array.isArray(deprecatedFunctions)) {
```

Comment:
I think `Array.isArray` here will not be useful as the `deprecatedFunctions` structure is:
```
"deprecatedFunctions": {
    "5.0.0": [
        "f1": {...}
        "f2": {...}
    ],
    "3.0.0": [
       "f1": {...}
    ]
}
```